File: G51/clase6/profiling/ejemploperf.py
"""
Programa para calcular una lista de factoriales desde 0! hasta n!
Carlos Andres Delgado Saavedra
18 de septiembre de 2024
"""

def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n-1)
    
def lista_factoriales(n):
    return [factorial(i) for i in range(n+1)]

if __name__ == "__main__":
    print(lista_factoriales(26))

File: G51/clase6/profiling/ejemploperf.cpp
/**
 * Programa en C++ para calcular el producto punto entre dos vectores
 * Carlos A Delgado S
 * 18 de Septiembre de 2024
 */

#include <iostream>
#include <vector>

using namespace std;

vector <int> v1;
vector <int> v2;

void llenarVectores(int n){
    for(int i = 0; i < n; i++){
        v1.push_back(i);
        v2.push_back(i);
    }
}

int productoPunto(int n){
    int resultado = 0;
    for(int i = 0; i < n; i++){
        resultado += v1[i] * v2[i];
    }
    return resultado;
}


int main(){
    int n = 1000000;
    llenarVectores(n);
    cout << "El producto punto es: " << productoPunto(n) << endl;
    return 0;
}